arangod/Cluster/Maintenance/Maintenance.cpp

```diff
@@ -294,6 +296,37 @@ void handlePlanShard(uint64_t planIndex, VPackSlice const& cprops, VPackSlice co
           LEADER_PRIORITY));
     }
 
+    if (shouldBeLeading) {
```

Comment:
Why this check? This does not look right. Rather, I would have expected a check if an upgrade is actually necessary, otherwise we would be creating such job descriptions rather frequently.

Reply:
The upgrade action is necessary if:
1) This is the shard leader and
2a) The Plan entry has an `upgradeStatus` value or
2b) The Current entry has an `upgradeStatus` value

If the concern is that we don't want to re-add the action here if one already exists, then that's a fair concern. I'm not sure how to do that though, thoughts?